dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Switch.mo

```diff
@@ -0,0 +1,32 @@
+within Dynawo.NonElectrical.Blocks.NonLinear;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, a hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block Switch "Switches two discrete inputs based on an xor of two booleans, one of which may change during the simulation"
+  import Dynawo.Connectors;
+  import Modelica.Blocks.Interfaces;
+
+  Connectors.ZPin doSwitch(value(start = -1)) "Whether a switch should occur or not. The filtering is done based on the discrete values of an ACMC: 0 undefined, 1 open, 2,3,4,5 closed" annotation(Placement(visible = true, transformation(origin = {-110, 70}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {-160, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  discrete Interfaces.RealInput u1 "One of the discrete inputs";
```

Comment:
Please add also an annotation for u1 and u2 + review the placement of the blocks in the diagram view and icon view.